pkg/binder/controller/pod_group_state_machine.md

```diff
@@ -1,3 +1,40 @@
 # Pod Group State Machine
 
-![pod group state machine](../../../docs/pics/pod_group_state_machine.jpg)
\ No newline at end of file
+```mermaid
+---
+title: PodGrop DFA
+---
+	%%{ init: { 'fontFamily': 'monospace', 'flowchart': { 'curve': 'cardinal' } } }%%
+flowchart TD
+  Pending --created >= min--> PreScheduling
+  PreScheduling -.created < min.-> Pending
+	Pending -.scheduled >= min.-> Scheduled
+	PreScheduling --scheduled >= min--> Scheduled
+	Pending --timeout--> Timeout
+	PreScheduling --timeout--> Timeout
+	
+	subgraph finalState[Final State]
+		Scheduled
+		Timeout
+	end
+	
+	style Timeout fill:#faa,color:black,font-weight:bold,stroke-width:2px,stroke:grey
+	style Scheduled fill:#dff,color:black,font-weight:bold,stroke-width:2px,stroke:grey
+	
+	style finalState fill:#f8ffff,stroke:#333,stroke-width:1px,color:#,stroke-dasharray: 5 5
+	style Pending fill:#d2dfff
+	style PreScheduling fill:#d2dfff
+```
+
+**Explanation**: Based on the definition of [DFA](https://en.wikipedia.org/wiki/Deterministic_finite_automaton), let us use `State` to represent $Q$ and use `Msg` to represent $\Sigma$, then we have the following state-transition table.
+
+| State \ Msg   | created < min | created >= min && scheduled < min | created >= min && scheduled >= min | timeout |
+| ------------- | ------------- | --------------------------------- | ---------------------------------- | ------- |
+| Pending       | -             | PreScheduling                     | Scheduled (Unexpected)             | Timeout |
```

Comment:
It's marked 'created >= min && scheduled >= min' as Unexpected here. Does it mean it's impossible to see a direct change of state from Pending to Scheduled; instead, it should always be Pending -> PreScheduled -> Scheduled?

Reply:
Yes, it's impossible to see a direct change of state from Pending to Scheduled.

Reply:
Eg. podgroup controller didn't handle the podgroup immediately due to RateLimitWorkQueue, which means its phase may not be PreScheduling when the binder binding the unit pods.